### PR TITLE
Add new spring boot starter module

### PR DESCRIPTION
**Pull request by eddumelendez**
This new module add all the configuration needed to integrate web
projects with sentry.

### CODE REVIEW

File: sentry-spring-boot-starter/pom.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <artifactId>sentry-all</artifactId>
+    <groupId>io.sentry</groupId>
+    <version>1.7.15-SNAPSHOT</version>
```
Comment (bretthoerner) [inline sentry-spring-boot-starter/pom.xml]: Can you rebase `master `and update this to `1.7.16-SNAPSHOT`?

File: sentry-spring-boot-starter/pom.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      <artifactId>spring-boot-autoconfigure</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-web</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.sentry</groupId>
+      <artifactId>sentry-spring</artifactId>
+      <version>1.7.15-SNAPSHOT</version>
```
Comment (bretthoerner) [inline sentry-spring-boot-starter/pom.xml]: Same note about version here.

File: sentry-spring-boot-starter/src/main/java/io/sentry/spring/autoconfigure/SentryAutoConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.web.servlet.ServletContextInitializer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.HandlerExceptionResolver;
+
+/**
+ * @author Eddú Meléndez
+ */
```
Comment (bretthoerner) [inline sentry-spring-boot-starter/src/main/java/io/sentry/spring/autoconfigure/SentryAutoConfiguration.java]: We don't have `@author` tags in the codebase, please remove.

File: sentry-spring-boot-starter/src/test/java/io/sentry/spring/autoconfigure/SentryAutoConfigurationTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import org.springframework.boot.autoconfigure.web.EmbeddedServletContainerAutoConfiguration;
+import org.springframework.boot.autoconfigure.web.HttpMessageConvertersAutoConfiguration;
+import org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext;
+import org.springframework.boot.test.util.EnvironmentTestUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Eddú Meléndez
+ */
```
Comment (bretthoerner) [inline sentry-spring-boot-starter/src/test/java/io/sentry/spring/autoconfigure/SentryAutoConfigurationTest.java]: Same note.

**Comment by eddumelendez** [comment]
@bretthoerner fixed :)

**Comment by bretthoerner** [comment]
Thanks @eddumelendez

**Pull request merged**